File: client-norx/app/src/main/java/ua/in/meugen/myapplication/ui/activities/base/BaseActivityModule.java
package ua.in.meugen.myapplication.ui.activities.base;

import android.content.Context;
import android.support.v7.app.AppCompatActivity;

import dagger.Binds;
import dagger.Module;
import ua.in.meugen.myapplication.app.di.qualifiers.ActivityContext;
import ua.in.meugen.myapplication.app.di.scopes.PerActivity;

@Module
public abstract class BaseActivityModule {

    @Binds @PerActivity
    abstract AppCompatActivity bindCompatActivuity(final BaseActivity activity);

    @Binds @ActivityContext @PerActivity
    abstract Context bindContext(final AppCompatActivity activity);
}

File: client-coroutines/app/src/main/java/ua/meugen/android/client/coroutines/ui/activities/base/BaseActivityModule.kt
package ua.meugen.android.client.coroutines.ui.activities.base

import android.content.Context
import android.support.v7.app.AppCompatActivity
import dagger.Binds
import dagger.Module
import ua.meugen.android.client.coroutines.app.di.PerActivity
import ua.meugen.android.client.coroutines.ui.lifecycle.LifecycleHandler
import ua.meugen.android.client.coroutines.ui.lifecycle.LoaderLifecycleHandler
import javax.inject.Named

const val ACTIVITY_CONTEXT = "activityContext"

@Module
abstract class BaseActivityModule {

    @Binds @Named(ACTIVITY_CONTEXT) @PerActivity
    abstract fun bindContext(activity: AppCompatActivity): Context
}